networkx/utils/misc.py

```diff
@@ -455,16 +460,18 @@ def __init__(self, rng=None):
             self._rng = rng
 
     def random(self):
-        return self._rng.random_sample()
+        return self._rng.random()
 
     def uniform(self, a, b):
-        return a + (b - a) * self._rng.random_sample()
+        return a + (b - a) * self._rng.random()
 
     def randrange(self, a, b=None):
+        if isinstance(self._rng, np.random.Generator):
+            return self._rng.integers(a, b)
         return self._rng.randint(a, b)
 
     def choice(self, seq):
-        return seq[self._rng.randint(0, len(seq))]
+        return self._rng.choice(seq)
```

Comment:
Both `RandomState` and `Generator` already have a `choice` method with consistent API, so there's no need to manually implement it. Fortunately the implementation for `RandomState.choice` is consistent with this:

https://github.com/numpy/numpy/blob/05d908a31c0be9db3177a5e2f5a543cbeca7e4f9/numpy/random/mtrand.pyx#L962

... so this change should be completely backward compatible.

Reply:
Ah - the test suite caught a difference: `np.random.RandomSequence.choice` only supports array_likes, which doesn't include strings, whereas `random.choice` supports all sequences.

I'll have to back this out and replace with an `isinstance` check for the `Generator` and `RandomState` branches.